examples/d3d12-visualization/src/main.rs

```diff
@@ -0,0 +1,556 @@
+#![windows_subsystem = "windows"]
+
+use raw_window_handle::HasRawWindowHandle;
+
+use gpu_allocator::d3d12::{Allocator, AllocatorCreateDesc};
+
+mod all_dxgi {
+    pub use winapi::shared::{
+        dxgi::*, dxgi1_2::*, dxgi1_3::*, dxgi1_4::*, dxgi1_6::*, dxgiformat::*, dxgitype::*,
+    };
+}
+
+use winapi::um::d3d12::*;
+use winapi::um::d3dcommon::*;
+use winapi::um::winuser;
+
+use winapi::shared::minwindef::UINT;
+use winapi::shared::windef::HWND;
+use winapi::shared::winerror;
+use winapi::shared::winerror::{FAILED, SUCCEEDED};
+
+use winapi::Interface;
+
+mod imgui_renderer;
+use imgui_renderer::{handle_imgui_event, ImGuiRenderer};
+
+const ENABLE_DEBUG_LAYER: bool = true;
+const FRAMES_IN_FLIGHT: usize = 2;
+
+struct BackBuffer {
+    resource: *mut ID3D12Resource,
+    rtv_handle: D3D12_CPU_DESCRIPTOR_HANDLE,
+}
+
+fn find_hardware_adapter(
+    dxgi_factory: &all_dxgi::IDXGIFactory6,
+) -> Option<*mut all_dxgi::IDXGIAdapter4> {
+    let mut adapter: *mut all_dxgi::IDXGIAdapter4 = std::ptr::null_mut();
+    let mut adapater_index = 0;
+    loop {
```

Comment:
```suggestion
    for adapter_index in 0..
```

Drop the `+=` in multiple places below.